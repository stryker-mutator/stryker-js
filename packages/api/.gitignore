/*.js
/*.d.ts
!/stryker.conf.js
testResources/module/**/*.js
testResources/module/**/*.d.ts