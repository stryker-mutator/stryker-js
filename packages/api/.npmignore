stryker.conf.js
test
dist/test
/schema
/reports
/coverage
/testResources
/.nyc_output
.vscode
.nycrc.json
*.tsbuildinfo
tsconfig?(.*).json
.mocharc.jsonc
.nyc_output
.gitattributes