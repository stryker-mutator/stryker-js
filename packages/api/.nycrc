{
    "nyc":{
        "lines": 100,
        "statements": 100,
        "functions": 100,
        "branches": 100,
        "exclude": [
            "testResources/**/*.js"
        ]
    }
}