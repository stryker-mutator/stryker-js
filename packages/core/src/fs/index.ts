export * from './file-system.js';
export * from './project.js';
export * from './project-file.js';
export * from './project-reader.js';
