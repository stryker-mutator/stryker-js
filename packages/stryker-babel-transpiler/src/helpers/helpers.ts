import * as path from 'path';

export function toJSFileName(fileName: string) {
  const fileNameInfo = path.parse(fileName);
  const newFileName = path.join(fileNameInfo.dir, fileNameInfo.name + '.js');
  return newFileName;
}
