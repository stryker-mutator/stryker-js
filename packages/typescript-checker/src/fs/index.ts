export * from './hybrid-file-system';
