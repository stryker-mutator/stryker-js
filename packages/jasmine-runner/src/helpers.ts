import { TestResult, TestStatus } from '@stryker-mutator/api/test-runner';
import jasmine from 'jasmine';

export const helpers = {
  toStrykerTestResult(specResult: jasmine.SpecResult, timeSpentMs: number): TestResult {
    const baseResult = {
      id: specResult.id.toString(),
      name: specResult.fullName,
      timeSpentMs,
    };
    if (specResult.status === 'disabled' || specResult.status === 'pending' || specResult.status === 'excluded') {
      return {
        ...baseResult,
        status: TestStatus.Skipped,
      };
    } else if (!specResult.failedExpectations || specResult.failedExpectations.length === 0) {
      return {
        ...baseResult,
        status: TestStatus.Success,
      };
    } else {
      return {
        ...baseResult,
        status: TestStatus.Failed,
        failureMessage: specResult.failedExpectations.map((failedExpectation) => failedExpectation.message).join(','),
      };
    }
  },
  createJasmine(options: jasmine.JasmineOptions): jasmine {
    return new jasmine(options);
  },
};
