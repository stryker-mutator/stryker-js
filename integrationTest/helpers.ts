import * as path from 'path';
import * as fs from 'mz/fs';
import { expect } from 'chai';
import { ScoreResult } from 'stryker-api/report';

export async function readScoreResult(eventResultDirectory = path.resolve('reports', 'mutation', 'events')) {
    const allReportFiles = await fs.readdir(eventResultDirectory);
    const scoreResultReportFile = allReportFiles.find(file => !!file.match(/.*onScoreCalculated.*/));
    expect(scoreResultReportFile).ok;
    const scoreResultContent = await fs.readFile(path.resolve(eventResultDirectory, scoreResultReportFile || ''), 'utf8');
    return JSON.parse(scoreResultContent) as ScoreResult;
}