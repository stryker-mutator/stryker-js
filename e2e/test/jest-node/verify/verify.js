import { promises as fsPromises } from 'fs';

import { fileURLToPath, URL } from 'url';

import { expect } from 'chai';

import { expectMetricsJsonToMatchSnapshot } from '../../../helpers.js';

describe('After running stryker with test runner jest on test environment "node"', () => {
  it('should report 75% mutation score', async () => {
    await expectMetricsJsonToMatchSnapshot();
  });
  it('should report test files and test locations', async () => {
    const sumTestFileName = fileURLToPath(new URL('../src/sum.test.js', import.meta.url));
    const report = JSON.parse(await fsPromises.readFile(fileURLToPath(new URL('../reports/mutation/mutation.json', import.meta.url)), 'utf-8'));
    const expectedTestFiles = {
      ['src/sum.test.js']: {
        source: await fsPromises.readFile(sumTestFileName, 'utf-8'),
        tests: [
          {
            id: '0',
            name: 'adds 1 + 2 to equal 3',
            location: { start: { line: 3, column: 2 } }, // columns should be 1, but for some reason aren't. Best guess if jest? ü§∑‚Äç‚ôÇÔ∏è
          },
          {
            id: '1',
            name: 'sub 1 - 0 to equal 1',
            location: { start: { line: 6, column: 2 } },
          },
        ],
      },
    };
    expect(report.testFiles).deep.eq(expectedTestFiles);
  });
});
