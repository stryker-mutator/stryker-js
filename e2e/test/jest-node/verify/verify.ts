import { expect } from 'chai';
import path from 'path';
import { promises as fsPromises } from 'fs';
import { MutationTestResult, TestFileDefinitionDictionary } from 'mutation-testing-report-schema/api';
import { expectMetricsJsonToMatchSnapshot } from '../../../helpers';

describe('After running stryker with test runner jest on test environment "node"', () => {
  it('should report 75% mutation score', async () => {
    await expectMetricsJsonToMatchSnapshot();
  });

  it('should report test files and test locations', async () => {
    const report: MutationTestResult = require('../reports/mutation/mutation.json');
    const expectedTestFiles: TestFileDefinitionDictionary = {
      [path.resolve(__dirname, '..', 'src', 'sum.test.js')]: {
        source: await fsPromises.readFile(path.resolve(__dirname, '..', 'src', 'sum.test.js'), 'utf-8'),
        tests: [{
          id: '0',
          name: 'adds 1 + 2 to equal 3',
          location: { start: { line: 3, column: 2 } } // columns should be 1, but for some reason aren't. Best guess if jest? ü§∑‚Äç‚ôÇÔ∏è
        }, {
          id: '1',
          name: 'sub 1 - 0 to equal 1',
          location: { start: { line: 6, column: 2 } }
        }]
      }
    };
    expect(report.testFiles).deep.eq(expectedTestFiles);
  });
});
