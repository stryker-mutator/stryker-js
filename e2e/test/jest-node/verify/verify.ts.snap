// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`After running stryker with test runner jest on test environment "node" should report 75% mutation score 1`] = `
Object {
  "compileErrors": 0,
  "ignored": 0,
  "killed": 3,
  "mutationScore": 75,
  "mutationScoreBasedOnCoveredCode": 75,
  "noCoverage": 0,
  "runtimeErrors": 0,
  "survived": 1,
  "timeout": 0,
  "totalCovered": 4,
  "totalDetected": 3,
  "totalInvalid": 0,
  "totalMutants": 4,
  "totalUndetected": 1,
  "totalValid": 4,
}
`;
